disnake/subscription.py

```diff
@@ -0,0 +1,122 @@
+from .mixins import Hashable
+from .utils import parse_time, snowflake_time
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.subscription import Subscription as SubscriptionPayload
+    from .user import User
+
+__all__ = ("Subscription",)
+
+
+class Subscription(Hashable):
+    """Represents a subscription.
+
+    This can only be retrieved using :meth:`SKU.subscriptions` or :meth:`SKU.fetch_subscription`.
+
+    .. warning::
+        :class:`Subscription`\\s should not be used to grant perks. Use :class:`Entitlement`\\s as a way of whether a user should have access to a specific :class:`SKU`.
+
+    .. note::
+        Some subscriptions may have been canceled already; consider using :meth:`is_canceled` to check whether a given subscription was canceled.
+
+    .. collapse:: operations
+
+        .. describe:: x == y
+
+            Checks if two :class:`Subscription`\\s are equal.
+
+        .. describe:: x != y
+
+            Checks if two :class:`Subscription`\\s are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the subscription's hash.
+
+    .. versionadded:: 2.10
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The subscription's ID.
+    user_id: :class:`int`
+        The ID of the user who is subscribed to the :attr:`sku_ids`.
+
+        See also :attr:`user`.
+    sku_ids: List[:class:`int`]
+        The ID of the SKUs the user is subscribed to.
+    renewal_sku_ids: List[:class:`int`]
+        The ID of the SKUs that will be renewed at the start of the new period.
+    entitlement_ids: List[:class:`int`]
+        The ID of the entitlements the user has.
+    current_period_start: :class:`datetime.datetime`
+        The time at which the current period for the given subscription started.
+    current_period_end: :class:`datetime.datetime`
+        The time at which the current period for the given subscription will end.
+    status: :class:`SubscriptionStatus`
+        The current status of the given subscription.
+    canceled_at: Optional[:class:`datetime.datetime`]
+        The time at which the subscription was canceled.
+
+        See also :attr:`is_canceled`.
+    """
+
+    __slots__ = (
+        "_state",
+        "id",
+        "user_id",
+        "sku_ids",
+        "entitlement_ids",
+        "renewal_sku_ids",
+        "current_period_start",
+        "current_period_end",
+        "status",
+        "canceled_at",
+    )
+
+    def __init__(self, *, data: SubscriptionPayload, state: ConnectionState) -> None:
+        self._state: ConnectionState = state
+
+        self.id: int = int(data["id"])
+        self.user_id: int = int(data["user_id"])
+        self.sku_ids: List[int] = list(map(int, data["sku_ids"]))
+        self.entitlement_ids: List[int] = list(map(int, data["entitlement_ids"]))
+        self.renewal_sku_ids: Optional[List[int]] = (
+            list(map(int, data["renewal_sku_ids"])) if data["renewal_sku_ids"] else None
+        )
```

Comment:
```suggestion
        self.renewal_sku_ids: Optional[List[int]] = (
            list(map(int, renewal_sku_ids))
            if (renewal_sku_ids := data.get("renewal_sku_ids")) is not None
            else None
        )
```
Just in case this is ever missing (despite only being documented as nullable)